### PR TITLE
Tsconfig param

### PR DESCRIPTION
**Pull request by AndrewLeedham**
Adds testing, docs and slightly better error handling to @apiel 's PR: https://github.com/vega/ts-json-schema-generator/pull/109

### CODE REVIEW

File: test/invalid-data.test.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ function assertSchema(name: string, type: string, message: string) {
     return () => {
         const config: Config = {
             path: resolve(`test/invalid-data/${name}/*.ts`),
+            tsconfig: undefined,
```
Comment (domoritz) [inline test/invalid-data.test.ts]:
```suggestion
```

File: test/valid-data.test.ts [hunk 2]
```diff
@@ -18,8 +18,8 @@ function assertSchema(
     return () => {
         const config: Config = {
             path: resolve(`${basePath}/${relativePath}/*.ts`),
+            tsconfig: undefined,
```
Comment (domoritz) [inline test/valid-data.test.ts]:
```suggestion
```

**Comment by AndrewLeedham** [comment]
> Thank you for the PR. Looks great. Can you make sure the tests pass on the CI?

I have been trying to fix them locally, but I am not really sure what is causing them. They don't seem really to the tsconfig changes.

**Comment by domoritz** [comment]
Do you get errors locally when you run the tests? Do you get the same errors on master?

**Comment by AndrewLeedham** [comment]
> Do you get errors locally when you run the tests? Do you get the same errors on master?

Errors are happening locally yes, and they don't happen on master. It is definitely something in this PR, I'm just not sure what yet.

**Comment by AndrewLeedham** [comment]
Figured it out, the tests are using the tsconfig.json for this project because of the findUp implementation, rather than the default config.

**Comment by domoritz** [comment]
> Figured it out, the tests are using the tsconfig.json for this project because of the findUp implementation, rather than the default config.

Makes me wonder whether we should only look for the tsconfig file between the current working directory and the provided file? Or disable search for tsconfig files completely and instead require an exact link to the tsconfig file. What do you think makes the most sense?

**Comment by AndrewLeedham** [comment]
Personally I think just allowing a file to be specified and not doing a find up would be fine. The find up behaviour is kind of unexpected and as I found hard to debug.

**Comment by domoritz** [comment]
I'm okay with requiring an exact path as well. Let's go with that.

**Comment by domoritz** [comment]
Thank you for the PR!

**Pull request merged**